Source/RxStarscream.swift

```diff
@@ -7,111 +7,112 @@ import RxSwift
 import RxCocoa
 import Starscream
 
+
 public enum WebSocketEvent {
-  case connected
-  case disconnected(NSError?)
-  case message(String)
-  case data(Foundation.Data)
-  case pong
+    case connected
+    case disconnected(Error?)
+    case message(String)
+    case data(Data)
+    case pong
 }
 
-public class RxWebSocketDelegateProxy: DelegateProxy,
-                                       WebSocketDelegate,
-                                       WebSocketPongDelegate,
-                                       DelegateProxyType {
-
-    public static func setCurrentDelegate(_ delegate: AnyObject?, toObject object: AnyObject) {
-        let webSocket = object as? WebSocket
-        webSocket?.delegate = delegate as? WebSocketDelegate
-        webSocket?.pongDelegate = delegate as? WebSocketPongDelegate
-    }
-
-    public static func currentDelegateFor(_ object: AnyObject) -> AnyObject? {
-        let webSocket = object as? WebSocket
-        return webSocket?.delegate
-    }
+public class RxWebSocketDelegateProxy: DelegateProxy<WebSocket, NSObjectProtocol>, DelegateProxyType, WebSocketDelegate, WebSocketPongDelegate {
 
     private weak var forwardDelegate: WebSocketDelegate?
     private weak var forwardPongDelegate: WebSocketPongDelegate?
 
     fileprivate let subject = PublishSubject<WebSocketEvent>()
 
-    required public init(parentObject: AnyObject) {
-        let webSocket = parentObject as? WebSocket
-        self.forwardDelegate = webSocket?.delegate
-        self.forwardPongDelegate = webSocket?.pongDelegate
-        super.init(parentObject: parentObject)
+    required public init(websocket: WebSocket) {
+        super.init(parentObject: websocket, delegateProxy: RxWebSocketDelegateProxy.self)
+    }
+
+    public static func currentDelegate(for object: WebSocket) -> NSObjectProtocol? {
+        return object.delegate as? NSObjectProtocol
+    }
+
+    public static func setCurrentDelegate(_ delegate: NSObjectProtocol?, to object: WebSocket) {
+        object.delegate = delegate as? WebSocketDelegate
+        object.pongDelegate = delegate as? WebSocketPongDelegate
+    }
+
+    public static func registerKnownImplementations() {
+        self.register { RxWebSocketDelegateProxy(websocket: $0) }
     }
 
-    public func websocketDidConnect(socket: WebSocket) {
-        subject.on(.next(WebSocketEvent.connected))
+    public func websocketDidConnect(socket: WebSocketClient) {
```

Comment:
Not entirely sure if this matter, but the type change here comes from Starscream itself? 
If yes - maybe the signature should change to `websocketDidConnect(client: _)`. Not entirely sure though. What do you think ?

Reply:
https://github.com/daltoniam/Starscream/commit/f506be54ae320bbe0703e005df5e943df0bf58a4#diff-04c6e90faac2675aa89e2176d2eec7d8

According to their documentation this changed with the Swift 4 update

Reply:
About the change, I'm not quite sure about it because it could lead people to create an instance of a WebSocketClient instead of the normal WebSocket instance (for example: socket = WebSocket(url: URL(string: "ws://localhost:8080/")!)), which didn't change.

Reply:
Gotya. Thanks !

Reply:
Don't worry about the change. The original project didn't change so there's no reason for us to.